pilot/pkg/model/push_context.go

```diff
@@ -947,3 +947,12 @@ func (ps *PushContext) initAuthorizationPolicies(env *Environment) error {
 	}
 	return nil
 }
+
+// AddVirtualServiceForTesting adds a virtual service to the push context.
+// It is to be used for TESTING ONLY.
+func (ps *PushContext) AddVirtualServiceForTesting(config *Config) {
```

Comment:
This is a little suspicious. Not familiar enough with Go though - is this the right way to do this?

Reply:
Nevermind I see this was already approved for master